src/ActiveLogin.Authentication.BankId.AspNetCore/Areas/BankIdAuthentication/Views/BankId/_LoginScript.cshtml

```diff
@@ -116,6 +120,12 @@
                         if (data.checkStatus) {
                             checkStatus(requestVerificationToken, returnUrl, loginOptions, data.orderRef);
                         }
+
+                        cancelButtonElement.addEventListener('click', function (event) {
+                            event.preventDefault();
+                            cancel(requestVerificationToken, data.orderRef);
+                            cancelButtonElement.onclick = undefined;
```

Comment:
Realised this won't work now.
Would suggest:
```js
if(!loginIsCancelledByUser) {
    cancel(requestVerificationToken, data.orderRef);
}
```

Should have the same result, right?

Reply:
Hmm, I do not think I understand which context you are referring to.

Reply:
`cancelButtonElement.onclick = undefined;`

Since we are using `cancelButtonElement.addEventListener('click'` setting `onclick ` to undefined wont work. There is the `once` option, but that's not supported in IE (https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) so would like to avoid that.

Checking the `loginIsCancelledByUser` is probably the easiest to not cancel multiple times.

Reply:
I see! I thought I saw the `onclick = undefined` elsewhere in the code. Perhaps I was mistaken. It should work as you say.

Reply:
Ahhh, yes, you are right. I will fix that in another PR.

Reply:
Ok, so a last shot.

What do you think about something like this:

```
var onCancel = function(event) {
    cancel(requestVerificationToken, data.orderRef);
    event.target.removeEventListener('click', onCancel);
};
cancelButtonElement.addEventListener('click', onCancel);
```

It is more verbose but perhaps it is more "javascripty"?

Reply:
Sounds great! Would just rename it to `onCancelButtonClick`